src/components/Menu/SelectMenu.tsx

```diff
@@ -0,0 +1,81 @@
+/* react-aria hook props types are inexplicitly typed */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { AriaListBoxOptions, useListBox, useListBoxSection, useOption } from "@react-aria/listbox";
+import { SelectState } from "@react-stately/select";
+import { ReactElement, useRef } from "react";
+import { AriaList } from "./Aria/AriaList";
+import { AriaMenuItem } from "./Aria/AriaMenuItem";
+import { AriaSection } from "./Aria/AriaSection";
+import { getKeyItemRecord, getMenuItems } from "./Aria/helper";
+import { MenuItemProps } from "./MenuItem/MenuItem";
+
+export type MenuItemType = MenuItemProps & { id: string };
+
+export type MenuBlock = {
+    id: string;
+    menuItems: MenuItemType[];
+};
+
+export type SelectMenuProps = {
+    menuBlocks: MenuBlock[];
+    ariaProps: AriaListBoxOptions<any>;
+    state: SelectState<any>;
+    ariaLabel?: string;
+};
+
+export const SelectMenu = ({
+    menuBlocks,
+    ariaProps,
+    state,
+    ariaLabel = "Select Menu",
+}: SelectMenuProps): ReactElement<SelectMenuProps> => {
+    const items = getMenuItems(menuBlocks);
+    const keyItemRecord = getKeyItemRecord(items);
+    const menuRef = useRef<HTMLUListElement | null>(null);
+    const { listBoxProps } = useListBox(
+        {
+            ...ariaProps,
+            "aria-label": ariaLabel,
+        },
+        state,
+        menuRef,
+    );
+
+    return (
+        <AriaList ariaProps={listBoxProps} ref={menuRef}>
+            {[...state.collection].map((section) => {
+                const sectionKey = section.key;
+                const { itemProps, groupProps } = useListBoxSection({ "aria-label": `${sectionKey}` });
+
+                return (
+                    <AriaSection key={sectionKey} sectionProps={itemProps} groupProps={groupProps}>
+                        {[...section.childNodes].map((item) => {
```

Comment:
does this need to be a copy of childNodes?

Reply:
It does according to the examples. Could be that it's typed incorrectly on the part of react-area.